Controller/DrawingObject/FieldLineDrawing.py

```diff
@@ -17,107 +20,46 @@ def draw(self, painter):
-            painter.drawArc(QtCore.QRect(x_rad_goal - rad_width,
-                                         y_rad_goal - rad_width,
-                                         rad_width * 2,
-                                         rad_width * 2),
-                            180 * 16, 90 * 16)
+            # Dessine tous les arcs
+            for name, arc in QtToolBox.field_ctrl.field_arcs.items():
+                self.drawArc(painter, arc)
+            for name, line in QtToolBox.field_ctrl.field_lines.items():
+                self.drawLine(painter, line)
 
+            # Dessine left goal
+            painter.setPen(QtToolBox.create_pen(color=BLUE_COLOR,
+                                                style='SolidLine',
+                                                width=50 * QtToolBox.field_ctrl.ratio_screen))
+            for name, line in QtToolBox.field_ctrl.field_goal_left.items():
+                self.drawLine(painter, line)
 
-            # Ligne de la surface de réparation (droite)
-            goal_line_resized = QtToolBox.field_ctrl.defense_stretch * QtToolBox.field_ctrl.ratio_screen
-            x1 = x + QtToolBox.field_ctrl.defense_radius * QtToolBox.field_ctrl.ratio_screen
-            y1 = y + width / 2 - goal_line_resized / 2
-            y2 = y + width / 2 + goal_line_resized / 2
-            painter.drawLine(x1, y1, x1, y2)
-            x1 = x + length - QtToolBox.field_ctrl.defense_radius * QtToolBox.field_ctrl.ratio_screen
-            painter.drawLine(x1, y1, x1, y2)
-
-            # Ligne de demi-terrain
-            x1, y1 = x + length / 2, y
-            x2, y2 = x + length / 2, y + width
-            painter.drawLine(x1, y1, x2, y2)
-
-            # Ligne centrale longitudinale (d'un but à l'autre)
-            x1, y1 = x, y + width/2
-            x2, y2 = x + length,  y + width/2
-            painter.drawLine(x1, y1, x2, y2)
-
-            # Ligne penality 1
-            ratio = QtToolBox.field_ctrl.ratio_screen
-            dist_from_goal_line = QtToolBox.field_ctrl.penalty_line_from_spot_dist + QtToolBox.field_ctrl.penalty_spot_from_field_line_dist
-            x1, y1 = x + dist_from_goal_line * ratio, y + width / 2 - 5  # TODO : faire selon la vision
-            x2, y2 = x + dist_from_goal_line * ratio, y + width / 2 + 5
-            painter.drawLine(x1, y1, x2, y2)
-
-            # Ligne penality 2
-            x1, y1 = x + length - dist_from_goal_line * ratio, y + width / 2 - 5  # TODO : faire selon la vision
-            x2, y2 = x + length - dist_from_goal_line * ratio, y + width / 2 + 5
-            painter.drawLine(x1, y1, x2, y2)
-
-            # Cercle central
-            radius = QtToolBox.field_ctrl.center_circle_radius * QtToolBox.field_ctrl.ratio_screen
-            x, y = x + length / 2, y + width / 2
-            painter.drawEllipse(x - radius, y - radius, radius * 2, radius * 2)
+            # Dessine right goal
+            painter.setPen(QtToolBox.create_pen(color=YELLOW_COLOR,
```

Comment:
Can we create a collection of pen to reuse them more consistently ? like
```
BOLD_SOLID_YELLOW
THIN_SOLID_YELLOW
```

Reply:
The variable 'QtToolBox.field_ctrl.ratio_screen' changes when you zoom in or out. So we can not really reuse them except if only reset them at every frame.